functions/HttpTriggerAPIUsersIdTask/__init__.py

```diff
@@ -1,24 +1,67 @@
 import logging
-
+import pyodbc
+import os
 import azure.functions as func
+import datetime
+import json
+import time
+
 
 
+# to handle datetime with JSON
+def default(o):
+    if isinstance(o, (datetime.datetime, datetime.date)):
+        return o.isoformat()
+
 def main(req: func.HttpRequest) -> func.HttpResponse:
-    logging.info('Python HTTP trigger function processed a request.')
+    
+
+    # define the server and database names
+    server = os.environ["ENV_DATABASE_SERVER"]
+    database = os.environ["ENV_DATABASE_NAME"]
+    username = os.environ["ENV_DATABASE_USERNAME"]
+    password = os.environ["ENV_DATABASE_PASSWORD"]
+    
+    # define the connection string
+    driver = '{ODBC Driver 17 for SQL Server}'
+    cnxn = "Driver={};Server={};Database={};Uid={};Pwd={};Encrypt=yes;TrustServerCertificate=no;Connection Timeout=30;".format(
+        driver, server, database, username, password)
+    
+   
+    id = req.params.get('userId')
 
-    name = req.params.get('name')
-    if not name:
+    if not id:
+        return func.HTTPResponse('Bad Request', status_code=400 )
+    else:
         try:
-            req_body = req.get_json()
-        except ValueError:
-            pass
+            with pyodbc.connect(cnxn) as conn:
+                return tasks_query(conn, id)
+        except pyodbc.DatabaseError as err:
+            if err.args[0] == '28000':
+                return func.HttpResponse(
+                    "Unauthorized User",
+                    status_code=403 
+                )
+
+# create the connection cursor
+def tasks_query(conn, id):
+    # create the query
+    with conn.cursor() as cursor:
+        cursor.execute(
+            "SELECT * FROM tasks WHERE userId={}".format(id))
+        row = cursor.fetchone()
+        if not row:
+            return func.HttpResponse(
+                "User not found",
+                status_code=404
+            )
         else:
-            name = req_body.get('name')
+            columns = [column[0] for column in cursor.description]

```

Comment:
As with @theKunte it'd be nice to have a comment of what's going on here.

Reply:
@kemet I think it would good to add your resource to it and explain what's going on. I know what's going on and I should have explained it a bit better. Good documentation is important! Great learning experience